qt-models/divepicturemodel.cpp

```diff
@@ -48,6 +51,29 @@ void DivePictureModel::updateDivePicturesWhenDone(QList<QFuture<void>> futures)
 	updateDivePictures();
 }
 
+void DivePictureModel::setZoomLevel(int level)
+{
+	zoomLevel = level / 10.0;
+	if (zoomLevel < -1.0)
+		zoomLevel = -1.0;
+	if (zoomLevel > 1.0)
+		zoomLevel = 1.0;
+	updateThumbnails();
+	layoutChanged();
+}
+
+void DivePictureModel::updateThumbnails()
+{
+	// Calculate size of thumbnails.
+	// Default is defaultIconMetrics().sz_pic.
+	// Minimum (zoomLevel == -1.0) is a third of that.
+	// Maximum (zoomLevel == 1.0) thrice that.
```

Comment:
OK, here's the explanation that I missed up there. I still don't understand why the zoom levels are in that range instead of 0...5 or something like that...

Reply:
I also extended the comment for this case:

@sfuchs79 requested scaling between 1/3rd and 3 times the standard sizes. If we scale linearly, then the central point would be 5/3rd of the standard size and not the standard size. Therefore the scaling is done exponentially and then -1=min, 0=standard, +1=max comes very naturally. Of course we could use [0..5] as range and then map that onto [-1 +1], but it seems to me that this would be much harder to follow? Perhaps it is just my mind which is diseased by working with exp() too much (https://de.wikipedia.org/wiki/Strukturfaktor) :°°(.

In the end, I think this is a matter of tuning and probably best done by someone with good tastes regarding UI (i.e. not me).

Reply:
That's the explanation that's missing in the code :-)
With that I'm good.